core/map.lua

```diff
@@ -313,6 +317,35 @@ function MinimapPinMixin:UpdateRotation()
     self.texture:SetRotation(self.rotation + math.pi * 2 - self.provider.facing)
 end
 
+function MinimapPinMixin:OnMouseEnter()
+    if self.label then
+        local x, _ = self:GetCenter()
+        local parentX, _ = self:GetParent():GetCenter()
+        if (x > parentX) then
+            GameTooltip:SetOwner(self, 'ANCHOR_LEFT')
+        else
+            GameTooltip:SetOwner(self, 'ANCHOR_RIGHT')
+        end
+
+        ns.PrepareLinks(self.label)
+        ns.PrepareLinks(self.note)
+
+        C_Timer.After(0, function()
+            -- label
+            GameTooltip:SetText(ns.RenderLinks(self.label, true))
+
+            -- note
+            if self.note and ns:GetOpt('show_notes') then
+                GameTooltip:AddLine(ns.RenderLinks(self.note), 1, 1, 1, true)
+            end
+
+            GameTooltip:Show()
+        end)
+    end
```

Comment:
This code is indentical for both minimap pins and map pins. Can we move it to a shared function that is called here? Something like this:

```lua
local function RenderPinTooltip(pin)
    ...
end

function MinimapPinMixin:OnMouseEnter()
    if self.label then
        RenderPinTooltip(self)
    end
end

function WorldMapPinMixin:OnMouseEnter()
    if self.label then
        RenderPinTooltip(self)
    end
end
```
```

Reply:
I moved this.